tests/e2e/framework/kubernetes.go

```diff
@@ -0,0 +1,248 @@
+package framework
+
+import (
+	"bufio"
+	"flag"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/golang/glog"
+	"istio.io/istio/tests/e2e/util"
+)
+
+const (
+	VERSION_FILE = "istio.VERSION"
+	YAML_SUFFIX  = ".yaml"
+)
+
+var (
+	kube               KubeInfo
+	istioctlUrlDefault string
+	mixerHubDefault    string
+	mixerTagDefault    string
+	managerHubDefault  string
+	managerTagDefault  string
+)
+
+type KubeInfo struct {
+	Namespace        string
+	NamespaceCreated bool
+	Hub              string
+	Tag              string
+	MixerImage       string
+	ManagerImage     string
+	CaImage          string
+	ProxyHub         string
+	ProxyTag         string
+	IstioctlUrl      string
+	Istioctl         string
+	Verbosity        int
+
+	TmpDir  string
+	YamlDir string
+
+	Ingress string
+}
+
+func init() {
+	parseVersion()
+
+	hubDefault := "gcr.io/istio-testing"
+	tagDefault := "b121a1e169365865e01a9e6eea066a34a29d9fd1"
+	var verbose bool
+
+	flag.StringVar(&kube.Hub, "hub", hubDefault, "Docker hub")
+	flag.StringVar(&kube.Tag, "tag", tagDefault, "Docker tag")
+	flag.StringVar(&kube.Namespace, "n", "", "Namespace to use for testing (empty to create/delete temporary one)")
+	flag.StringVar(&kube.MixerImage, "mixer", mixerHubDefault+"/mixer:"+mixerTagDefault, "Mixer image")
+	flag.StringVar(&kube.ManagerImage, "manager", managerHubDefault+"/manager:"+managerTagDefault, "Manager image")
+	flag.StringVar(&kube.CaImage, "ca", "", "Ca image")
+	flag.StringVar(&kube.ProxyHub, "proxy_hub", managerHubDefault, "proxy hub")
+	flag.StringVar(&kube.ProxyTag, "proxy_tag", managerTagDefault, "proxy tag")
+	flag.StringVar(&kube.IstioctlUrl, "istioctl_url", istioctlUrlDefault, "URL to download istioctl")
+	flag.BoolVar(&verbose, "verbose", false, "Debug level noise from proxies")
+
+	if verbose {
+		kube.Verbosity = 3
+	} else {
+		kube.Verbosity = 2
+	}
+}
+
+func NewKubeInfo(tmpDir, runId string) *KubeInfo {
```

Comment:
This you call this new, you might want to do a copy of it and send the copy.
var k = new(KubeInfo)
*k = kube
and do everything on k